monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -546,11 +545,14 @@ def report_is_outdated(cls) -> bool:
         return bool(cls._agent_repository.get_agents())
 
     @classmethod
-    def get_report(cls):
+    def update_report(cls):
         if cls._agent_repository is None:
             raise RuntimeError("Agent repository does not exists")
 
         if cls.report_is_outdated():
-            cls._report = safe_generate_regular_report()
+            with cls._report_generation_lock:
```

Comment:
We should lock before "report_is_outdated()". Otherwise we'll unnecessarily regenerate reports.